packages/big-design/src/components/Worksheet/RowStatus/RowStatus.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { memo, useMemo } from 'react';
+
+import { useStore } from '../hooks';
+
+import { Status } from './styled';
+
+interface RowStatusProps {
+  rowIndex: number;
+}
+
+export const RowStatus: React.FC<RowStatusProps> = memo(({ rowIndex }) => {
+  const isSelected = useStore(useMemo(() => (state) => state.selectedRows.includes(rowIndex), [rowIndex]));
+
+  const isInvalid = useStore(
+    useMemo(() => (state) => Boolean(state.invalidCells.find((cell) => cell.rowIndex === rowIndex)), [rowIndex]),
```

Comment:
🍹  This line may benefit from a `.some()` as well since it already returns a boolean value.